### DISCUSSION TITLE
Repeatable blocks for rich web pages

### DISCUSSION THREAD

**Discussion by Kainkainkain** [directus/directus#4768] [Archive]
I can't figure out if this is possible, I've been looking at relationship fields but doesn't seem to be possible. 

**Use Case Explanation**
So I would like users to be able to edit rich pages within the CMS. These pages are made of blocks - so imagine a hero block, a call to action block, a recent news block, a full width text block. 

- A hero block for example might contain the fields: "Left or right?", "Hero title", "Byline", "Background image", "Background image size". 
- Further, those blocks may contain predefined blocks. Imagine 'add button' block within the hero. 
- Some CMSs call this a repeater field. 

Has anyone else had success building a data model like this?

**Comment by Kainkainkain** [comment]
Found my own answer: Yes. just had to look at more relationship options. 
**Many-to-Any (M2A)**

**Reply by liamb13** [reply]
Could you explain your solution a little more?
I'm currently trying to work out the same thing

I somewhat understand how to do the blocks, but I can't seem to work out how to rearrange them if needed

**Reply by Kainkainkain** [reply]
https://docs.directus.io/concepts/relationships/
I read that - but I haven't had the time to test it - re-arranging blocks would most likely be something I want to do so if it doesn't have some sort of draggable sort then it might be a no-go for me. 

It's a shame as I can get more out of forestry, netlifyCMS and "smaller" cms's without fuss.

**Reply by rijkvanzanten** [reply]
> re-arranging blocks would most likely be something I want to do so if it doesn't have some sort of draggable sort then it might be a no-go for me.

The m2a setup supports drag-and-drop natively if you add a sort field 👍🏻

**Reply by Kainkainkain** [reply]
Ah there we go, it will work.
